tasmota/tasmota_xsns_sensor/xsns_115_wooliis.ino

```diff
@@ -0,0 +1,230 @@
+  }
+  if (WooliisSerial->available() < 21) {
+    return;
+  }
+  uint8_t buffer[21];
+  WooliisSerial->readBytes(buffer, 21);
+  if (buffer[0] != 0xb5 && buffer[1] != 0x5b && buffer[2] == 0x01 && buffer[3] == 0x01) {
+    AddLog(LOG_LEVEL_DEBUG, PSTR("Wooliis: Flushing serial"));
+    WooliisSerial->flush();  // Out of sync or wrong sensor connected
+    AddLogBuffer(LOG_LEVEL_DEBUG_MORE, buffer, 21);
+    return;
+  }
+ 
+  uint8_t checksum = 0;
+  for (uint32_t i = 0; i < 20; i++) {
+    checksum += buffer[i];
+  }
+  checksum = 255 - checksum;
+  if (checksum != buffer[20]) {
+    AddLog(LOG_LEVEL_DEBUG, PSTR("Wooliis: " D_CHECKSUM_FAILURE));
+    return;
+  }
+
+  AddLog(LOG_LEVEL_DEBUG, PSTR("Wooliis: message received correctly!"));
+  AddLogBuffer(LOG_LEVEL_DEBUG_MORE, buffer, 21);
+
+  Wooliis.charge_percent = buffer[4];
+  Wooliis.remaining_capacity = 0.1f * ((uint16_t)buffer[5]<<8 | buffer[6]);
+  Wooliis.voltage = 0.1f * ((uint16_t)buffer[7]<<8 | buffer[8]);
+  Wooliis.current = (buffer[19] ? -0.1f : 0.1f) * ((uint16_t)buffer[9]<<8 | buffer[10]);
+  Wooliis.energy_in = 0.1f * ((uint32_t)buffer[11]<<16 | (uint16_t)buffer[12]<<8 | buffer[13]);
+  Wooliis.energy_out = 0.1f * ((uint32_t)buffer[14]<<16 | (uint16_t)buffer[15]<<8 | buffer[16]);
+  Wooliis.status = buffer[19];
+
+  Wooliis.valid = 1;
+
+  return;
+}
+
+/*********************************************************************************************/
+
+void WooliisSecond(void)                 // Every second
+{
+  if (Wooliis.ready)
+    WooliisReadData();
+}
+
+/*********************************************************************************************/
+
+void WooliisInit(void) {
+  if (PinUsed(GPIO_WOOLIIS_RX)) {
+    WooliisSerial = new TasmotaSerial(Pin(GPIO_WOOLIIS_RX), -1, 1);
+    if (WooliisSerial->begin(9600)) {
+      if (WooliisSerial->hardwareSerial()) { ClaimSerial(); }
+      WooliisSerial->flush();  // Clear serial buffer
+      Wooliis.ready = 1;
+      AddLog(LOG_LEVEL_DEBUG, PSTR("Wooliis: Serial ready"));
+    }
+  }
+  Wooliis.valid = 0;
+}
+
+#ifdef USE_WEBSERVER
+const char HTTP_SNS_WOOLIIS_BCP[] = "{s}%s " D_BATTERY_CHARGE "{m}%d " D_UNIT_PERCENT "{e}";
+const char HTTP_SNS_WOOLIIS_CAP[] = "{s}%s " D_CAPACITY "{m}%s " D_UNIT_CHARGE "{e}";
+const char HTTP_SNS_WOOLIIS_CHG[] = "{s}%s " D_CHARGING "{m}%s{e}";
+const char HTTP_SNS_WOOLIIS_VOL[] = "{s}%s " D_VOLTAGE "{m}%s " D_UNIT_VOLT "{e}";
+const char HTTP_SNS_WOOLIIS_CUR[] = "{s}%s " D_CURRENT "{m}%s " D_UNIT_AMPERE "{e}";
+const char HTTP_SNS_WOOLIIS_POW[] = "{s}%s " D_POWERUSAGE "{m}%s " D_UNIT_WATT "{e}";
+const char HTTP_SNS_WOOLIIS_EIN[] = "{s}%s " D_ENERGY " " D_INGRESS "{m}%s " D_UNIT_WATTHOUR "{e}";
+const char HTTP_SNS_WOOLIIS_EEG[] = "{s}%s " D_ENERGY " " D_EGRESS "{m}%s " D_UNIT_WATTHOUR "{e}";
+#endif  // USE_WEBSERVER
+
+void WooliisShow(bool json) {
+  char float_str[8];
+  if (Wooliis.valid) {
+    if (json) {
+      ResponseAppend_P(PSTR(",\"%s\":{"), D_WOOLIIS);
```

Comment:
Do not use translated text in english only JSON messages. Use already available D_JSON_xxx defines and if needed add your own english only D_JSON_xxx to your local file.